x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -238,6 +245,28 @@ func (m *MongoCrypt) setLocalSchemaMap(schemaMap map[string]bsoncore.Document) e
 	return nil
 }
 
+// setEncryptedFieldsMap sets the encryptedfields map in mongocrypt.
+func (m *MongoCrypt) setEncryptedFieldsMap(encryptedfieldsMap map[string]bsoncore.Document) error {
+	if len(encryptedfieldsMap) == 0 {
+		return nil
+	}
+
+	// convert encryptedfields map to BSON document
+	midx, mdoc := bsoncore.AppendDocumentStart(nil)
+	for key, doc := range encryptedfieldsMap {
+		mdoc = bsoncore.AppendDocumentElement(mdoc, key, doc)
+	}
+	mdoc, _ = bsoncore.AppendDocumentEnd(mdoc, midx)
```

Comment:
Do we need to handle this error?

Reply:
This has been replaced with a call to `bson.Marshal`. The error returned is checked.